src/Diagnostics.DataProviders/ArmClient.cs

```diff
@@ -7,22 +7,24 @@ namespace Diagnostics.DataProviders
     internal class ArmClient : IGeoMasterClient
     {
         private const string CsmEndpointUrl = "https://management.azure.com/";
-        public HttpClient Client { get; }
-        public Uri BaseUri { get; }
+        private static readonly HttpClient _client = new HttpClient();
+        public HttpClient Client => _client;
+        public Uri BaseUri => _client.BaseAddress;
+        public string AuthenticationToken { get; }
 
-        public ArmClient(GeoMasterDataProviderConfiguration configuration)
+        static ArmClient()
         {
-            var handler = new HttpClientHandler();
-            BaseUri = new Uri(CsmEndpointUrl);
+            _client.BaseAddress = new Uri(CsmEndpointUrl);
+            _client.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+        }
 
-            Client = new HttpClient(handler)
+        public ArmClient(GeoMasterDataProviderConfiguration configuration)
+        {
+            if (configuration.Token == null)
```

Comment:
Can we use this check: string.IsNullOrWhiteSpace(configuration.Token) to cover the empty string or whitespace as well?

Reply:
Yes, that's apparently better. I will update.

Reply:
Updated